src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs

```diff
@@ -1646,6 +1646,29 @@ void M()
             }
         }
 
+        [Fact]
+        public void CS0221ERR_ConstOutOfRangeCheckedNullableInt()
+        {
+            // Confirm that we truncate the constant value before performing the range check
+            var template =
+@"public class C
+{
+    void M()
+    {
+        System.Console.WriteLine((int?)int.MinValue);
+        System.Console.WriteLine((int?)double.MinValue); //CS0221
+        System.Console.WriteLine((int?)int.MaxValue);
+        System.Console.WriteLine((int?)double.MaxValue); //CS0221
+    }
+}
+";
+
+            CreateCompilation(template)
+                .VerifyDiagnostics(
+                    Diagnostic(ErrorCode.ERR_ConstOutOfRangeChecked, "(int?)double.MinValue").WithArguments("double", "int?").WithLocation(6, 34),
```

Comment:
>double [](start = 109, length = 6)

The name of a type for the first fill-in does not make sense in the context of this diagnostic.

Reply:
I changed to 
```csharp
 new ErrorDescription { Code = (int)ErrorCode.ERR_ConstOutOfRangeChecked, Line = 15, Column = 13  }
```
or it's better keep the original diagnostic? And replace the values:

```csharp
Diagnostic(ErrorCode.ERR_ConstOutOfRangeChecked, "(int?)double.MinValue").WithArguments("1.7976931348623157E+308", "int?").WithLocation(6, 34),
```